dataset.py

```diff
@@ -137,3 +138,41 @@ def partitioned_reference_data(reference_csv_filename: str) -> PartitionedDatase
   print_split(partition_data)
   return partition_data
 
+
+def preprocess_sample_data(df, feature_columns, label_columns, aggregate_columns, keep_grouping) -> pd.DataFrame:
```

Comment:
naming the parameters is probably a good idea.  optionally having default for keep_grouping.